damnit/gui/editor.py

```diff
@@ -123,6 +121,94 @@ def __init__(self):
+            self.setMatchedBraceBackgroundColor(QColor('#404040'))
+            self.setMatchedBraceForegroundColor(QColor('white'))
+            self.setUnmatchedBraceBackgroundColor(QColor('#802020'))
+            self.setUnmatchedBraceForegroundColor(QColor('white'))
+            
+            # Python syntax highlighting colors for dark theme
+            self._lexer.setDefaultPaper(QColor('#232323'))
+            self._lexer.setDefaultColor(QColor('white'))
+            self._lexer.setColor(QColor('#66d9ef'), QsciLexerPython.Keyword)
+            self._lexer.setColor(QColor('#a6e22e'), QsciLexerPython.ClassName)
+            self._lexer.setColor(QColor('#f92672'), QsciLexerPython.Operator)
+            self._lexer.setColor(QColor('#fd971f'), QsciLexerPython.FunctionMethodName)
+            self._lexer.setColor(QColor('#75715e'), QsciLexerPython.Comment)
+            self._lexer.setColor(QColor('#e6db74'), QsciLexerPython.DoubleQuotedString)
+            self._lexer.setColor(QColor('#e6db74'), QsciLexerPython.SingleQuotedString)
+            self._lexer.setColor(QColor('#ae81ff'), QsciLexerPython.Number)
+            
+            # Set paper (background) for all styles
+            for style in range(128):  # QScintilla uses style numbers 0-127
+                self._lexer.setPaper(QColor('#232323'), style)
+        else:
+            # Light theme colors (default)
+            self.setPaper(QColor('white'))
+            self.setColor(QColor('black'))
+            self.setCaretForegroundColor(QColor('black'))
+            self.setCaretLineBackgroundColor(QColor('lightgray'))
+            self.setMarginsForegroundColor(QColor('black'))
+            self.setMarginsBackgroundColor(QColor('white'))
+            self.setSelectionBackgroundColor(QColor('#c0c0c0'))
+            self.setSelectionForegroundColor(QColor('black'))
+            self.setMatchedBraceBackgroundColor(QColor('#c0c0c0'))
+            self.setMatchedBraceForegroundColor(QColor('black'))
+            self.setUnmatchedBraceBackgroundColor(QColor('#ff8080'))
+            self.setUnmatchedBraceForegroundColor(QColor('black'))
+            
+            # Python syntax highlighting colors for light theme
+            self._lexer.setDefaultPaper(QColor('white'))
+            self._lexer.setDefaultColor(QColor('black'))
+            self._lexer.setColor(QColor('#0000ff'), QsciLexerPython.Keyword)
+            self._lexer.setColor(QColor('#007f7f'), QsciLexerPython.ClassName)
+            self._lexer.setColor(QColor('#7f0000'), QsciLexerPython.Operator)
+            self._lexer.setColor(QColor('#007f00'), QsciLexerPython.FunctionMethodName)
+            self._lexer.setColor(QColor('#7f7f7f'), QsciLexerPython.Comment)
+            self._lexer.setColor(QColor('#7f007f'), QsciLexerPython.DoubleQuotedString)
+            self._lexer.setColor(QColor('#7f007f'), QsciLexerPython.SingleQuotedString)
+            self._lexer.setColor(QColor('#007f7f'), QsciLexerPython.Number)
```

Comment:
Do we want to have all the colours defined in theme.py, and then have this method do something like:

```python
colours = get_highlighting_colours(self.current_theme)

...
self._lexer.setColor(colours['number'], QsciLexerPython.Number)
```
```

Up to you - I don't feel strongly about how we define the colours, it just seemed worth asking the question.

Reply:
Good point, I moved the color definition to `theme.py`